File: 2023/01/b.cpp
#include <bits/stdc++.h>
using namespace std;

#define bug(x) cout << #x << "=" << x << '\n'

int get_number(string line) {
  int first_digit = 0;
  for (int i = 0; i < line.length(); ++i) {
    char c = line[i];
    if (isdigit(c)) {
      first_digit = int(c - '0');
      break;
    }
  }

  int last_digit = 0;
  for (int i = line.length() - 1; i >= 0; --i) {
    char c = line[i];
    if (isdigit(c)) {
      last_digit = int(c - '0');
      break;
    }
  }

  int number = first_digit * 10 + last_digit;
  return number;
}

int main(int argc, char **argv) {
  for (int f = 1; f < argc; ++f) {
    cout << "\n===== " << argv[f] << " =====\n";
    int ans = 0;
    ifstream infile(argv[f]);

    string line;
    while (getline(infile, line)) {
      int number = get_number(line);
      ans += number;
    }

    bug(ans);
  }

  return 0;
}

File: 2023/01/b.py
# CORRECT FOR 2ND
import sys

NUMBER_NAMES = [
    "zero",
    "one",
    "two",
    "three",
    "four",
    "five",
    "six",
    "seven",
    "eight",
    "nine",
]


def get_number(input: str):
    first_digit = -1
    for i in range(len(input)):
        if first_digit != -1:
            break

        if input[i].isnumeric():
            first_digit = int(input[i])

        for j in range(len(NUMBER_NAMES)):
            if NUMBER_NAMES[j] in input[:i]:
                first_digit = j

    last_digit = -1
    for i in reversed(range(len(input))):
        if last_digit != -1:
            break

        if input[i].isnumeric():
            last_digit = int(input[i])

        for j in range(len(NUMBER_NAMES)):
            if NUMBER_NAMES[j] in input[i:]:
                last_digit = j

    print(f"{first_digit}, {last_digit} <== {input}")
    number = first_digit * 10 + last_digit
    return number


if __name__ == "__main__":
    input_files = sys.argv[1:]

    for input in input_files:
        print(f"\n===== {input} =====")
        with open(input) as f:
            input_data = f.readlines()

        answer = 0
        for i in input_data:
            answer += get_number(i)

        print("answer: ", answer)
